### PR TITLE
support output packet type

### PR DESCRIPTION
**Pull request by Asphaltt**
1. output pkt_type of skb
2. output CPU

Signed-off-by: Leon Huayra <hffilwlqm@gmail.com>

### CODE REVIEW

**Comment by Asphaltt** [comment]
Here's an output example:

```text
               SKB          PROCESS                     FUNC      CPU        TIMESTAMP
0xffff8f25c1127500   [tmux: server]             kfree_skbmem        3   45557282884169 netns=4026531992 mark=0x0 pkt_type=HOST ifindex=2 proto=8 mtu=1500 len=76 10.0.2.15:39917->8.8.8.8:53(udp)
0xffff8f26ce799800    [dbus-daemon]             kfree_skbmem        1   45558067343261 netns=4026531992 mark=0x0 pkt_type=HOST ifindex=0 proto=0 mtu=0 len=176 [7923:0:9c00:0:101:6f00:2300:0]:0->[2f6f:7267:2f66:7265:6564:6573:6b74:6f70]:0()
0xffff8f262df5b900         [vminfo]             kfree_skbmem        2   45558067607046 netns=4026531992 mark=0x0 pkt_type=HOST ifindex=0 proto=0 mtu=0 len=215 [7a23:0:7500:0:601:7300:400:0]:0->[3a31:2e37::401:7300:2900:0]:0()
```

File: bpf/kprobe_pwru.c [hunk 1]
```diff
@@ -59,6 +60,8 @@ struct event_t {
 	u64 addr;
 	u64 skb_addr;
 	u64 ts;
+	u32 cpu;
+	u32 pad;
```
Comment (brb) [inline bpf/kprobe_pwru.c:64]: Did you check with `pahole` that this padding is needed?

Comment (Asphaltt) [reply bpf/kprobe_pwru.c]: Great, `pahole` is the very tool that I need.

File: bpf/kprobe_pwru.c [hunk 2]
```diff
@@ -38,7 +38,8 @@ struct skb_meta {
 	u32 len;
 	u32 mtu;
 	u16 protocol;
-	u16 pad;
+	u8 pkt_type;
```
Comment (brb) [inline bpf/kprobe_pwru.c:41]: AFAIK, the packet type should be available in `--output-skb` (it should print it if it's non-zero). Have you checked it?

**Comment by Asphaltt** [comment]
I should close this PR. I cannot run `--output-skb`.

```bash
# ./pwru --output-skb
2021/11/25 16:47:00 Loading objects: field KprobeSkb1: program kprobe_skb_1: load program: invalid argument: BPF_STX uses reserved fields
processed 0 insns (limit 1000000) max_states_per_insn 0 total_states 0 peak_states 0 mark_read 0
```

**Comment by brb** [comment]
@Asphaltt Could you open the PR for outputting CPU?

> invalid argument: BPF_STX uses reserved fields

Which Linux kernel and Clang version are you using?

**Comment by Asphaltt** [comment]
PR #38  for outputing CPU

issue #39  for `--output-skb` failure.

**Pull request closed without merging**